BuildResidentialHPXML/measure.rb

```diff
@@ -3388,23 +3363,58 @@ def self.create(runner, model, args, epw_path)
     end
 
     # Check for errors in the HPXML object
-    if args[:apply_validation].is_initialized && args[:apply_validation].get
+    if args[:apply_validation]
       errors = hpxml.check_for_errors()
       if errors.size > 0
         fail "ERROR: Invalid HPXML object produced.\n#{errors}"
       end
     end
 
-    if args[:apply_defaults].is_initialized && args[:apply_defaults].get
-      eri_version = Constants.ERIVersions[-1]
-      HPXMLDefaults.apply(runner, hpxml, eri_version, weather, epw_file: epw_file)
-    end
-
     hpxml_doc = hpxml.to_oga()
+    XMLHelper.write_file(hpxml_doc, hpxml_path)
+
+    if args[:apply_validation] || args[:apply_defaults]
+
+      # Check for invalid HPXML file
+      if not validate_hpxml(runner, hpxml_path, hpxml_doc)
+        return false
+      end
+
+      if args[:apply_defaults]
+        eri_version = Constants.ERIVersions[-1]
+        HPXMLDefaults.apply(runner, hpxml, eri_version, weather, epw_file: epw_file)
+        hpxml_doc = hpxml.to_oga()
+      end
+    end
```

Comment:
Seems like this could be simplified to:

```suggestion
    if args[:apply_validation]
      # Check for invalid HPXML file
      if not validate_hpxml(runner, hpxml_path, hpxml_doc)
        return false
      end
    end

    if args[:apply_defaults]
      eri_version = Constants.ERIVersions[-1]
      HPXMLDefaults.apply(runner, hpxml, eri_version, weather, epw_file: epw_file)
      hpxml_doc = hpxml.to_oga()
    end
```